solidity/contracts/converter/BancorConverter.sol

```diff
@@ -528,44 +520,45 @@ contract BancorConverter is IBancorConverter, TokenHandler, SmartTokenController
       * @param _sourceToken source ERC20 token
       * @param _targetToken target ERC20 token
       * @param _amount      amount of tokens to convert (in units of the source token)
-      * @param _minReturn   if the conversion results in an amount smaller than the minimum return - it is cancelled, must be nonzero
       * @param _beneficiary wallet to receive the conversion result
       *
       * @return amount of tokens received (in units of the target token)
     */
-    function convertInternal(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount, uint256 _minReturn, address _beneficiary)
+    function convertInternal(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount, address _trader, address _beneficiary)
         public
         payable
         protected
         only(BANCOR_NETWORK)
-        greaterThanZero(_minReturn)
         returns (uint256)
     {
         require(_sourceToken != _targetToken); // validate input
 
+        // if a whitelist is set, verify that both and trader and the beneficiary are whitelisted
+        require(conversionWhitelist == address(0) ||
+                (conversionWhitelist.isWhitelisted(_trader) && conversionWhitelist.isWhitelisted(_beneficiary)));

```

Comment:
we can upgrade it to receive an array but since it's only relevant if whitelist is actually defined (0 such converters in the network right now), it's not worth it at this point.

Reply:
@yudilevi:
1. Also removes the `&&` (which might make it worth if this is called several times during a conversion-request by the user).
2. If we are planning to redeploy the white-list contract anyway, then it is definitely worth adding it IMO.

Reply:
that contract is deployed per converter that needs it and currently there's none so there's no deployment at the moment